lib/librarytest.go

```diff
@@ -0,0 +1,242 @@
+// +build e2e_test
+
+// This is a file with e2e tests for C bindings written in library.go.
+// As a CGO file, it can't have `_test.go` suffix as it's not allowed by Go.
+// At the same time, we don't want this file to be included in the binaries.
+// This is why `e2e_test` tag was introduced. Without it, this file is excluded
+// from the build. Providing this tag will include this file into the build
+// and that's what is done while running e2e tests for `lib/` package.
+
+// Additionaly this file should contain test that mock the Status API.
+// Existing test in 'utils.go' that test the Status API will be migrated to the
+// e2e package and test that test the C Binding will be migrated to this file
+
+package main
+
+/*
+#include <stdlib.h>
+*/
+import "C"
+
+import (
+	"fmt"
+	"testing"
+	"unsafe"
+
+	"github.com/golang/mock/gomock"
+	"github.com/status-im/status-go/geth/common"
+	"github.com/stretchr/testify/assert"
+)
+
+func testCreateAccountWithMock(t *testing.T) {
+	realStatusAPI := statusAPI
+	defer func() { statusAPI = realStatusAPI }()
+
+	// Setup Mock StatusAPI
+	ctrl := gomock.NewController(t)
+	status := NewMockStatusAPI(ctrl)
+	statusAPI = status
+	accountInfo1 := common.AccountInfo{Address: "add", Mnemonic: "mne", PubKey: "Pub"}
+	accountInfo2 := common.AccountInfo{Error: "Error Message"}
+	status.EXPECT().CreateAccount("pass1").Return(accountInfo1)
+	status.EXPECT().CreateAccount("").Return(accountInfo1)
+	status.EXPECT().CreateAccount(C.GoString(nil)).Return(accountInfo1)
+	status.EXPECT().CreateAccount("pass2").Return(accountInfo2)
+
+	// C Strings
+	pass1 := C.CString("pass1")
+	pass2 := C.CString("pass2")
+	empty := C.CString("")
+	jsonResult := C.CString(`{"address":"add","pubkey":"Pub","mnemonic":"mne","error":""}`)
+	jsonResultError := C.CString(`{"address":"","pubkey":"","mnemonic":"","error":"Error Message"}`)
+	defer func() {
```

Comment:
Is it needed?

Reply:
Assuming you mean the C.free(unsafe.Pointer(config1)). I think they are needed as C.CString are not cleaned up by GC 

Although as this is a short running test it may not matter if the strings are leaked.